### PR TITLE
Update feature freeze procedure

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

This updates the feature freeze procedure to reflect I think what we want reality to be.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/development/releasing.rst [hunk 1]
```diff
@@ -274,30 +277,43 @@ The procedure for this is straightforward:
 
       $ git branch v<version>.x
 
-#. Render the changelog with towncrier, as above.
```
Comment (astrofrog) [inline docs/development/releasing.rst:277]: As @Cadair mentioned, we (a) only want to do this for the final release (not at branching time or RCs etc), and (b) we should do it on the release branch and forward-port it to ``main``

File: docs/development/releasing.rst [hunk 2]
```diff
@@ -274,30 +277,43 @@ The procedure for this is straightforward:
...
+   point to the that version.
+
+#. Update the "what's new" section of the current version,
+  ``docs/whatsnew/<current_version>.rst``, and remove all content, replacing it
+  with::
+
+      :orphan:
 
+      `What's New in Astropy <current_version>?
+      <https://docs.astropy.org/en/v<current_version>/whatsnew/<current_version>.html>`__
```
Comment (astrofrog) [inline docs/development/releasing.rst:299]: One small issue with this is that it results in a temporarily broken link until v5.0 is actually released.

File: docs/development/releasing.rst (hunk 1 above)
Comment (eteq) [reply docs/development/releasing.rst:277]: Hmm I agree with (b) but not necessarily (a).  See my suggestion just below

File: docs/development/releasing.rst [hunk 3]
```diff
@@ -274,30 +277,43 @@ The procedure for this is straightforward:
...
-#. Render the changelog with towncrier, as above.
-
-#. Tag this commit using the next major version followed by ``.dev``. For example,
-   if you have just branched ``4.0``, create the ``v4.1.dev`` tag on the commit
-   adding the ``4.1`` section to the changelog::
-
-      $ git tag -s "v<next_version>.dev" -m "Back to development: v<next_version>"
-
-#. Also update the "what's new" section of the docs to include a section for the
+#. Update the "what's new" section of the docs to include a section for the
```
Comment (eteq) [inline docs/development/releasing.rst:280]:
```suggestion
#. Create a new branch from main at the point you want the feature freeze to
   occur and check it out::

      $ git checkout -b v<version>-changelog
      
#.  Render the changelog in the maintainience branch with towncrier, as above. 
      
#.  Create a PR with that changelog, for others to review. Changes in that branch should *not* be made it the rendered version, however, rather they should go in the fragments and later back-ported to main and the release branches.
      
#.  Switch back to the main branch::

      $ git checkout main

#. Update the "what's new" section of the docs to include a section for the
```
I'm not sure if this is the optimal solution, but *some* sort of review of the changelog should happen between now and the RC.

**Review by eteq — APPROVED** [review]
This looks right to me aside from maybe wanting to add one more step - see my suggestion.

Note I'm not sure if exactly this suggestion is right, and maybe we shouldn't have it hold up this PR anyway, so I'm approving on the theory we might just merge it and fix it in another PR.

File: docs/development/releasing.rst (hunk 3 above)
Comment (astrofrog) [reply docs/development/releasing.rst:280]: I'll open a separate PR for now with something along these lines but with a modification (just in the interest of not having one giant PR for all the release doc updates)

**Pull request merged**